### PR TITLE
Introduce nested limit checks

### PR DESCRIPTION
**Pull request by b10n1k**
Update .perlcriticrc to run nested limit checks and fixed issue for lib/registration.pm

- Related ticket: https://progress.opensuse.org/issues/48605
- Needles: N/A
- Verification run: https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/536122487
and http://aquarius.suse.cz/tests/949 shows that changes are not destructive

### CODE REVIEW

**Comment by b10n1k** [comment]
the Travis now is failing because of the introduced nested limits.

**Comment by b10n1k** [comment]
those are modules that should be assigned to the corresponding teams to take care of. i didnt make anything with them. @rwx788 do you want me to fix those?

**Comment by okurz** [comment]
Hi @b10n1k, I guess fixing these issues yourself is actually easier than waiting for someone else. For the "slenkins" file I am sure there is no other "team" currently that would feel better responsible. However, there would also be the alternative to exclude these two files explicitly from the checks. That sounds a bit messy though ;)

File: .perlcriticrc [hunk 1]
```diff
@@ -14,3 +14,8 @@ severity = 5
 
 [Perl::Critic::Policy::ConsistentQuoteLikeWords]
 severity = 5
+
+[ControlStructures::ProhibitDeepNests]
+severity = 4
+add_themes = freenode
```
Comment (jknphy) [inline .perlcriticrc]: Good catch here, according to documentation, add_themes makes more sense than set_themes even we could see same output.

**Comment by jknphy** [comment]
LGTM

**Pull request merged**